tools/myst/enc/enc.c

```diff
@@ -154,26 +136,38 @@ static uint64_t _forward_exception_as_signal_to_kernel(
         siginfo.si_signo = SIGSEGV;
         // `si_addr` is always null for #GP.
         siginfo.si_addr = NULL;
-        (*_kargs.myst_handle_host_signal)(&siginfo, &_mcontext);
-        _mcontext_to_oe_context(&_mcontext, oe_context);
-        return OE_EXCEPTION_CONTINUE_EXECUTION;
+        supported_code = true;
     }
     if (oe_exception_code == OE_EXCEPTION_X87_FLOAT_POINT)
     {
         // ATTN: Consider implementing accurate si-code for
         // OE_EXCEPTION_X87_FLOAT_POINT
         siginfo.si_code = FPE_FLTINV;
         siginfo.si_signo = SIGFPE;
-        (*_kargs.myst_handle_host_signal)(&siginfo, &_mcontext);
-        _mcontext_to_oe_context(&_mcontext, oe_context);
-        return OE_EXCEPTION_CONTINUE_EXECUTION;
+        supported_code = true;
     }
     if (oe_exception_code == OE_EXCEPTION_DIVIDE_BY_ZERO)
     {
         siginfo.si_code = FPE_INTDIV;
         siginfo.si_signo = SIGFPE;
-        (*_kargs.myst_handle_host_signal)(&siginfo, &_mcontext);
-        _mcontext_to_oe_context(&_mcontext, oe_context);
+        supported_code = true;
+    }
+
+    if (supported_code)
+    {
```

Comment:
It's better to add comment here to explain the design and its rationale. Something like - OE layer HW exception handling scheme requires any registered exception handler call to return, so it can perform cleanup after the return. Invoking any signal handler of the program language runtime or the application in the register exception handler here will not meet OE layer HW exception handling scheme requirement, as the signal handler of the program language runtime or the application might not return. Setting up oe_context and return to OE layer with OE_EXCEPTION_CONTINUE_EXECUTION, to "jump to" myst_handle_host_signal() using Mystikos stack, after OE layer finishes its HW exception handling cleanup.